lib/models/message.ts

```diff
@@ -233,31 +167,25 @@ function toBaseType(o: IRetrievedMessage): IMessage {
   }
 }
 
-function toRetrieved(
-  result: DocumentDb.RetrievedDocument
-): IRetrievedMessageWithContent | IRetrievedMessageWithoutContent {
-  if (isIRetrievedMessageWithContent(result)) {
-    return {
-      ...result,
-      kind: "IRetrievedMessageWithContent"
-    } as IRetrievedMessageWithContent;
-  } else {
-    // TODO: we kind of trusting the data storage that this is indeed valid
-    // TODO: possibly return an Option in case result fails validation
-    return {
-      ...result,
-      kind: "IRetrievedMessageWithoutContent"
-    } as IRetrievedMessageWithoutContent;
+function toRetrieved(result: DocumentDb.RetrievedDocument): RetrievedMessage {
+  if (RetrievedMessageWithContent.is(result)) {
+    return result;
+  }
+  if (RetrievedMessageWithoutContent.is(result)) {
+    return result;
   }
+  throw new Error(
```

Comment:
throw ? cant' we return an option or either as stated in the old comments ?

Reply:
this is actually supposed to never happen, as you can see the old code was just casting the object, I decided to avoid the case and make the validation explicit

yes we could have `toRetrieved` return an `Option` but I'm not sure the added complexity makes it worth it - anyway, this is outside of the scope of this PR